operator/pkg/util/patcher/pather.go

```diff
@@ -69,6 +76,9 @@ func (p *Patcher) ForDeployment(deployment *appsv1.Deployment) {
 	deployment.Annotations = labels.Merge(deployment.Annotations, p.annotations)
 	deployment.Spec.Template.Annotations = labels.Merge(deployment.Spec.Template.Annotations, p.annotations)
 
+	if p.resources.Size() > 0 {
+		deployment.Spec.Template.Spec.Containers[0].Resources = p.resources
```

Comment:
It'd be better to have a comment just like below:
> // It's considered the first container is the karmada component by default.

Reply:
done